### PR TITLE
Fix readthedocs builds

### LINKED ISSUE #96: The readthedocs builds fail
Here's a failed build log.

https://readthedocs.org/api/v2/build/12285883.txt

The salient bit appears to be:

```
  gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/docs/.pyenv/versions/2.7.18/include/python2.7 -c build/temp.linux-x86_64-2.7/_openssl.c -o build/temp.linux-x86_64-2.7/build/temp.linux-x86_64-2.7/_openssl.o
  build/temp.linux-x86_64-2.7/_openssl.c:873:6: error: conflicting types for ‘BIO_new_mem_buf’
   BIO *BIO_new_mem_buf(void *, int);
        ^~~~~~~~~~~~~~~
  In file included from /usr/include/openssl/asn1.h:65:0,
                   from build/temp.linux-x86_64-2.7/_openssl.c:581:
  /usr/include/openssl/bio.h:692:6: note: previous declaration of ‘BIO_new_mem_buf’ was here
   BIO *BIO_new_mem_buf(const void *buf, int len);
        ^~~~~~~~~~~~~~~
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for cryptography
```

### PR DESCRIPTION
**Pull request by exarkun**
Fixes #96

### CODE REVIEW

File: .requirements-readthedocs.txt [hunk 1]
```diff
@@ -1,35 +1,48 @@
...
-pytz==2015.7
-restructuredtext-lint==0.14.0
-six==1.10.0
-snowballstemmer==1.2.1
-sphinx-rtd-theme==0.1.9
-stevedore==1.10.0
-wheel==0.26.0
-zope.interface==4.1.3
+virtualenv==16.7.9
+zope.interface==5.2.0
```
Comment (exarkun) [inline .requirements-readthedocs.txt:48]: What I did was create a new virtualenv, pip install Imaginary[doc] into it, and then `pip freeze` into this file.

**Comment by altendky** [comment]
I don't get quite exactly the same thing for 2.7.18 in Linux.  Though some of the differences may be just `requests` having been released in the past week, maybe something else too.  `virtualenv` seems like a manual addition though?

<details> <summary>pip freeze</summary>

```
alabaster==0.7.12
appdirs==1.4.4
argon2-cffi==20.1.0
attrs==20.3.0
Automat==20.2.0
axiom==0.9.0
Babel==2.9.0
bcrypt==3.1.7
certifi==2020.11.8
cffi==1.14.3
chardet==3.0.4
constantly==15.1.0
cryptography==3.2.1
cssutils==1.0.2
docutils==0.16
enum34==1.1.10
epsilon==0.8.0
hyperlink==20.0.1
idna==2.10
imagesize==1.2.0
Imaginary @ git+https://github.com/twisted/imaginary@be87547a298af747c82b6bf878961f8d5a2e3714
incremental==17.5.0
ipaddress==1.0.23
Jinja2==2.11.2
Mantissa==0.9.0
MarkupSafe==1.1.1
Nevow==0.14.5
packaging==20.4
passlib==1.7.4
Pillow==6.2.2
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.20
Pygments==2.5.2
PyHamcrest==1.10.1
pyOpenSSL==19.1.0
pyparsing==2.4.7
pytz==2020.4
requests==2.25.0
service-identity==18.1.0
six==1.15.0
snowballstemmer==2.0.0
Sphinx==1.8.5
sphinxcontrib-websupport==1.1.2
Twisted==20.3.0
txpasslib==0.1.0
typing==3.7.4.3
urllib3==1.26.2
Vertex==0.3.1
zope.interface==5.2.0
```

</summary>

**Comment by exarkun** [comment]
> I don't get quite exactly the same thing for 2.7.18 in Linux. Though some of the differences may be just `requests` having been released in the past week, maybe something else too. `virtualenv` seems like a manual addition though?
> 
> pip freeze

Hm.  Yea virtualenv doesn't seem required, so deleted that.  The other jiggling is ... maybe an inevitable result of just installing the latest release of everything?  Though I wonder if it could also have to do with non-deterministic pip extra resolution.  Fortunately I think none of it matters.

**Review by altendky — APPROVED** [review]
The following almost matches and these versions were latest a week ago.

```
pip install certifi==2020.6.20 urllib3==1.25.11 Babel==2.8.0 requests==2.24.0 git+https://github.com/twisted/imaginary@96.readthedocs-build#egg=imaginary[doc]
```

My local is still including the following.  Didn't dig too hard with `pipdeptree`.

```
pyasn1-modules==0.2.8
service-identity==18.1.0
```

But you seem ok with this so alrighty.

**Pull request merged**